### PR TITLE
Update to Babel 7.x - WIP

### PR DESCRIPTION
**Pull request by benedyktdryl**
Hi, 

This is my very firs PR which includes Babel 7.x related updates. Please let me know if you will be interested in such contribution so I can make further updates in that matter.

---

This PR is also related to those PRs.

- https://github.com/lukeed/taskr/pull/305
- https://github.com/zeit/styled-jsx/pull/362

---

I know it has conflict but I don't know if you're accepting such contributions or you want to put some updates to Babel packages so I made this PR to ask first, before any further work.

### CODE REVIEW

**Review by przemyslaw-rzasa — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -49,23 +49,23 @@
...
+    "@babel/plugin-syntax-dynamic-import": "7.0.0-beta.32",
+    "@babel/plugin-transform-modules-commonjs": "7.0.0-beta.32",
+    "@babel/plugin-transform-react-jsx-source": "7.0.0-beta.32",
+    "@babel/plugin-transform-runtime": "7.0.0-beta.32",
+    "@babel/preset-env": "7.0.0-beta.32",
+    "@babel/preset-react": "7.0.0-beta.32",
+    "@babel/runtime": "7.0.0-beta.32",
+    "@babel/template": "7.0.0-beta.32",
     "ansi-html": "0.0.7",
-    "babel-core": "6.26.0",
```
Comment (haohcraft) [inline package.json]: Looks like [this](https://github.com/zeit/next.js/blob/canary/server/build/babel/find-config.js#L2) is still using `babel-core`.

**Comment by haohcraft** [comment]
Hope this PR will get merged soon.

**Comment by timneutkens** [comment]
It's on my list for Next v6. There's a few modules completely breaking down on babel 7. Will close this since I have a WIP locally.

File: package.json (hunk 1 above)
Comment (xtuc) [reply package.json]: You will need to change it `@babel/core`, but in general you should rely on Babel's internals. 

We already had a few discussions to expose the config to plugins authors https://github.com/babel/babel/pull/7259.

**Comment by xtuc** [comment]
@timneutkens could you address my concern https://github.com/zeit/next.js/pull/3428#discussion_r170526519?

File: package.json (hunk 1 above)
Comment (timneutkens) [reply package.json]: If that PR gets merged we'll change to it.

**Pull request closed without merging**